transformers/storage/median/repository.go

```diff
@@ -0,0 +1,286 @@
+package median
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/makerdao/vdb-mcd-transformers/transformers/shared"
+	"github.com/makerdao/vdb-mcd-transformers/transformers/shared/constants"
+	"github.com/makerdao/vdb-mcd-transformers/transformers/storage"
+	"github.com/makerdao/vdb-mcd-transformers/transformers/storage/utilities/wards"
+	"github.com/makerdao/vulcanizedb/libraries/shared/storage/types"
+	"github.com/makerdao/vulcanizedb/pkg/datastore/postgres"
+)
+
+const (
+	insertMedianValQuery  = `INSERT INTO maker.median_val (diff_id, header_id, address_id,  val) VALUES ($1, $2, $3, $4) ON CONFLICT DO NOTHING`
+	insertMedianAgeQuery  = `INSERT INTO maker.median_age (diff_id, header_id, address_id, age) VALUES ($1, $2, $3, $4) ON CONFLICT DO NOTHING`
+	insertMedianBarQuery  = `INSERT INTO maker.median_bar (diff_id, header_id, address_id, bar) VALUES ($1, $2, $3, $4) ON CONFLICT DO NOTHING`
+	insertMedianBudQuery  = `INSERT INTO maker.median_bud (diff_id, header_id, address_id, a, bud) VALUES ($1, $2, $3, $4, $5) ON CONFLICT DO NOTHING`
+	insertMedianOrclQuery = `INSERT INTO maker.median_orcl (diff_id, header_id, address_id, a, orcl) VALUES ($1, $2, $3, $4, $5) ON CONFLICT DO NOTHING`
+	insertMedianSlotQuery = `INSERT INTO maker.median_slot (diff_id, header_id, address_id, slot_id, slot) VALUES ($1, $2, $3, $4, $5) ON CONFLICT DO NOTHING`
+)
+
+type MedianStorageRepository struct {
+	db              *postgres.DB
+	ContractAddress string
+}
+
+func (repository MedianStorageRepository) Create(diffID, headerID int64, metadata types.ValueMetadata, value interface{}) error {
+	switch metadata.Name {
+	case storage.Packed:
+		return repository.insertPackedValueRecord(diffID, headerID, repository.ContractAddress, metadata, value.(map[int]string))
+	case wards.Wards:
+		return wards.InsertWards(diffID, headerID, metadata, repository.ContractAddress, value.(string), repository.db)
+	case Bar:
+		return repository.insertMedianBar(diffID, headerID, repository.ContractAddress, value.(string))
+	case Bud:
+		return repository.insertBud(diffID, headerID, metadata, repository.ContractAddress, value.(string))
+	case Orcl:
+		return repository.insertOrcl(diffID, headerID, metadata, repository.ContractAddress, value.(string))
+	case Slot:
+		return repository.insertSlot(diffID, headerID, metadata, repository.ContractAddress, value.(string))
```

Comment:
more nitpicking 😅 but why are we passing `repository.ContractAddress` in the args? Seems like these functions could just access that value directly

Reply:
Whoops. Overlooked that one. 😓